Adamant/Modules/Chat/ViewModel/ChatViewModel.swift

```diff
@@ -1425,7 +1425,9 @@ private extension ChatViewModel {
                 return
             }
         case .serverError:
-            dialog.send(.richError(error))
+            if error.localizedDescription.contains("Timestamp is in the future") {
+                dialog.send(.error(.adamant.alert.timeAheadError, supportEmail: false))
+            }
```

Comment:
Let's add an enum with possible server errors. `timestampInTheFuture` and `unknown(String)` for now. Make this enum an associated value for the `serverError`

Add mapping to `ServerResponse+Resolver.swift`

Reply:
Update: just add another case to the `ChatsProviderError` enum